### PR TITLE
feat: Added feature for creating a new geofence with a pre-defined square

### PR DESCRIPTION
**Pull request by thaddmt**
#### Description of changes
- Added feature for creating a geofence
- Clicking add button shows a prompt for naming the geofence/ setting the id
- Clicking add confirm button adds a predefined square geofence that to the screen which the developer can edit
- Gif below of the features added

Untitled

#### Checklist

- [x] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### CODE REVIEW

**Review by Milan-Shah — APPROVED** [review]

File: src/AmplifyGeofenceControl.ts [hunk 1]
```diff
@@ -100,36 +105,29 @@ export class AmplifyGeofenceControl {
...
     console.log(this._mapBoxDraw);
-    // console.log(
-    //   "Drew a polygon " +
-    //     (this._mapBoxDraw.getAll().features[0].geometry as any).coordinates
-    // );
+    console.log(
+      "Saving a polygon " +
+        (this._mapBoxDraw.getAll().features[0].geometry as any).coordinates
+    );
+    console.log(this._mapBoxDraw.get(this._editingGeofenceId));
```
Comment (TreTuna) [inline src/AmplifyGeofenceControl.ts:112]: Remove console logs?
```suggestion
```

File: src/AmplifyGeofenceControl.ts [hunk 2]
```diff
@@ -150,33 +148,33 @@ export class AmplifyGeofenceControl {
       },
     };
 
-    const renderGeofences = this._renderGeofences;
-    this._map.on("load", function () {
-      renderGeofences([hardcodeGeofence]);
+    // const renderGeofences = this._renderGeofences;
+    // this._map.on("load", function () {
+    //   renderGeofences([hardcodeGeofence]);
+    // });
```
Comment (TreTuna) [inline src/AmplifyGeofenceControl.ts:154]: Do we need this anymore?
```suggestion
```

File: src/utils.ts [hunk 3]
```diff
@@ -1,6 +1,9 @@
+import along from "@turf/along";
+import distance from "@turf/distance";
+import { lineString, point } from "@turf/helpers";
```
Comment (TreTuna) [inline src/utils.ts:3]: Looks like the only thing used from these imports is:
```suggestion
import { point } from "@turf/helpers";
```

File: src/utils.ts [hunk 4]
```diff
@@ -1,6 +1,9 @@
...
 import { Feature } from "geojson";
-import { AnySourceImpl, GeoJSONSource } from "maplibre-gl";
-import { Coordinates, NamedLocation, Geofence } from "./types";
+import { AnySourceImpl, GeoJSONSource, LngLatBounds } from "maplibre-gl";
+import { Coordinates, NamedLocation, Geofence, Polygon } from "./types";
```
Comment (TreTuna) [inline src/utils.ts:6]: I don't see either of these new imports being used in this file.
```suggestion
import { AnySourceImpl, GeoJSONSource } from "maplibre-gl";
import { Coordinates, NamedLocation, Geofence } from "./types";
```

**Review by TreTuna — CHANGES_REQUESTED** [review]
A few little things, mostly things that seem like they can be removed.

**Pull request merged**